### DISCUSSION TITLE
Semantic Token - Highlighting problem on multilines

### DISCUSSION THREAD

**Discussion by theogiraudet** [eclipse-langium/langium#524] [Q&A] [answered]
Hello,
I don't really know if it's a bug or not, so I prefer post here first.
In my model, I have a semantic element named "Documentation" that can be on multiple lines (as many languages do).
I have defined a SemanticTokenProvider in wich I do this:
```typescript
else if(isDocumentation(node)) {
    acceptor({ node, feature: "doc", type: SemanticTokenTypes.comment, modifier: SemanticTokenModifiers.documentation });
}
```
But when I test the result:
![image](https://user-images.githubusercontent.com/28688370/171829832-86afb16f-6f59-48df-8430-37366e5a1080.png)
The highlighting doesn't work when the documentation in on multiple lines.

Is there a bug or I doing something wrong?

Here is the code:
https://github.com/theogiraudet/Minimal-Langium-Semantic-Token-Issue

Thanks!

**Comment by msujew** [comment, marked as answer]
Hi @theogiraudet,

this isn't necessarily a bug nor intended behavior, but a vscode limitation. Language clients provide a boolean `multilineTokenSupport` for the `semanticTokens` request. If that's `false`, the client will discard any response data which tries to do semantic highlighting for multiple lines. Unfortunately, that's the case for vscode:

![grafik](https://user-images.githubusercontent.com/4377073/171841678-ae782a2c-10e5-47f5-98f3-2e7a36f2dd3c.png)

Generally, I would recommend you to use the textMate grammar support in that case. Alternatively, you can manually identify the cst node that belongs to the `doc` assignment (using the `findNodeForFeature` method) and generate a semantic token for each line that belongs to the comment node.

**Reply by theogiraudet** [reply]
Thanks for the solution! In my case, the targeted front isn't VSCode and doesn't natively support TextMate so I prefer the solution with `findNodeForFeature` to stay in the LSP protocole.

**Reply by msujew** [reply]
@theogiraudet By the way, with https://github.com/langium/langium/pull/553 the default semantic token provider will automatically split tokens into multiple lines (if the `multilineTokenSupport` is `false`). So for 0.5, your workaround won't be necessary anymore 👍

**Reply by theogiraudet** [reply]
Oh really nice! Thanks for the information ^^
